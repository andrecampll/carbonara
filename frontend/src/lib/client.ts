import { ApolloClient, HttpLink, InMemoryCache } from '@apollo/client'
import { registerApolloClient } from '@apollo/experimental-nextjs-app-support/rsc'

import { env } from '@/env'

export const { getClient } = registerApolloClient(() => {
  return new ApolloClient({
    cache: new InMemoryCache(),
    link: new HttpLink({
      uri: env.NEXT_PUBLIC_API_BASE_URL,
    }),
  })
})
