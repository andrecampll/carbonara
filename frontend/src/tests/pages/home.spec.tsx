import { MockedProvider } from '@apollo/client/testing'
import { cleanup, screen } from '@testing-library/react'
import { HTMLAttributes } from 'react'
import { afterEach, describe, expect, it, vitest } from 'vitest'

import HomePage from '@/app/(pages)/(home)/page'
import { render } from '@/utils/tests/render'

vitest.mock('next/image', () => ({
  __esModule: true,
  default: (props: HTMLAttributes<HTMLImageElement>) => {
    return <img {...props} />
  },
}))

afterEach(cleanup)

describe('HomePage', () => {
  it('should be able to render Home page correctly', () => {
    const { container } = render(
      <MockedProvider addTypename={false}>
        <HomePage />
      </MockedProvider>,
    )

    expect(
      screen.getByRole('heading', {
        name: /Lets Calculate Your Carbon Footprint/i,
      }),
    ).toBeDefined()
    expect(
      screen.getByText(
        'Follow the steps to calculate your carbon footprint. We will provide you with a detailed report at the end. Lets get started!',
      ),
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
