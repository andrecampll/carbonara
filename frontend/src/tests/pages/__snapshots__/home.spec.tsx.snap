// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomePage > should be able to render Home page correctly 1`] = `
<div
  class="my-6 flex w-full"
>
  <div
    class="w-full"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div
          style="z-index: 1; opacity: 1; transform: none;"
        >
          <div
            class="flex flex-col gap-4"
          >
            <header>
              <h5
                class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
              >
                Lets Calculate Your Carbon Footprint
              </h5>
              <p
                class="MuiTypography-root MuiTypography-body1 css-3sis8q-MuiTypography-root"
              >
                Follow the steps to calculate your carbon footprint. We will provide
      you with a detailed report at the end. Lets get started!
              </p>
            </header>
            <div
              class="my-4 flex flex-col items-center gap-4"
            >
              <img
                alt=""
                height="200"
                src="/images/carbon-light.webp"
                width="200"
              />
            </div>
            <footer
              class="flex w-full items-center justify-end"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary css-f76bo7-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="submit"
              >
                Next
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </footer>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`HomePage > should be able to render Home page with FootprintWizard component 1`] = `
<div
  class="my-6 flex w-full"
>
  <div
    class="w-full"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div
          style="z-index: 1; opacity: 1; transform: none;"
        >
          <div
            class="flex flex-col gap-4"
          >
            <header>
              <h5
                class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
              >
                Lets Calculate Your Carbon Footprint
              </h5>
              <p
                class="MuiTypography-root MuiTypography-body1 css-3sis8q-MuiTypography-root"
              >
                Follow the steps to calculate your carbon footprint. We will provide
      you with a detailed report at the end. Lets get started!
              </p>
            </header>
            <div
              class="my-4 flex flex-col items-center gap-4"
            >
              <img
                alt=""
                height="200"
                src="/images/carbon-light.webp"
                width="200"
              />
            </div>
            <footer
              class="flex w-full items-center justify-end"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary css-f76bo7-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="submit"
              >
                Next
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </footer>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
